internal/cmp/lb_profiles.go

```diff
@@ -115,15 +99,69 @@ func (lb *loadBalancerProfile) Create(ctx context.Context, d *utils.Data, meta i
 }
 
 func (lb *loadBalancerProfile) Delete(ctx context.Context, d *utils.Data, meta interface{}) error {
-	lbProfileID := d.GetID()
-	lbDetails, err := lb.lbClient.GetLoadBalancers(ctx)
-	if err != nil {
+	setMeta(meta, lb.lbClient.Client)
+	var tfLBProfile models.CreateLBProfileReq
+	if err := tftags.Get(d, &tfLBProfile); err != nil {
 		return err
 	}
-	_, err = lb.lbClient.DeleteLBProfile(ctx, lbDetails.GetNetworkLoadBalancerResp[0].ID, lbProfileID)
+
+	resp, err := lb.lbClient.DeleteLBProfile(ctx, tfLBProfile.LbID, tfLBProfile.ID)
 	if err != nil {
 		return err
 	}
 
+	if !resp.Success {
+		return fmt.Errorf("got success = 'false' while deleting LB-PROFILE")
+	}
+
 	return nil
 }
+
+func (lb *loadBalancerProfile) Update(ctx context.Context, d *utils.Data, meta interface{}) error {
+	id := d.GetID()
+
+	updateReq := models.CreateLBProfile{
+		CreateLBProfileReq: models.CreateLBProfileReq{
+			Name:        d.GetString("name"),
+			LbID:        d.GetInt("lb_id"),
+			Description: d.GetString("description"),
+			ServiceType: d.GetString("service_type"),
+			ProfileConfig: models.LBProfile{
+				ProfileType:            d.GetString("profile_type"),
+				RequestHeaderSize:      d.GetInt("request_header_size"),
+				ResponseHeaderSize:     d.GetInt("response_header_size"),
+				ResponseTimeout:        d.GetInt("response_timeout"),
+				HTTPIdleTimeout:        d.GetInt("http_idle_timeout"),
+				FastTCPIdleTimeout:     d.GetInt("fast_tcp_idle_timeout"),
+				ConnectionCloseTimeout: d.GetInt("connection_close_timeout"),
+				HaFlowMirroring:        d.GetBool("ha_flow_mirroring"),
+				CookieMode:             d.GetString("cookie_mode"),
```

Comment:
are these cookie related attributes collected from users as input?

Reply:
cookie mode input is optional by default it will take INSERT if we want to change it as PREFIX, REWRITE then user needs to give the value

Reply:
What about other attributes related to cookies

Reply:
cookie_type, cookie_fallback, cookie_garbling and cookie_mode having default values but  cookie_name is required filed if we select cookie from serviceType